### PR TITLE
change capture warning if merchant is eligible for ipp

### LINKED ISSUE #2921: [Mobile Payments] Update capture warning to indicate in-person payments must be captured in 48 hours
@laurendavissmith - please prioritize this text change highly

Right now we have this warning in settings for merchants who elect to auth without capture:

`Charge must be captured on the order details screen within 7 days of authorization, otherwise the authorization and order will be canceled.`

Xiao/Stripe noted today that, for in-person payments, authorizations must be captured within **48 hours**.

At a minimum, we should update that text to reflect that. Additional issues may be considered for warning merchants in wp-admin that they have un-captured authorizations nearing their expiration time, or disallowing auth-only for in-person payments. Feel free to comment below or open additional issues.

Edit: I am assuming this checkbox also controls auth/capture for in-person payments. I could be mistaken.

cc @juliacallicrate @astralbodies @ctarda 

Related: pdfdoF-9o-p2

must-be-captured

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #2921

#### Changes proposed in this Pull Request

Update capture warning to indicate in-person payments must be captured in 48 hours

If merchant is eligible for in-person payments, add a note to the "Issue an authorization on checkout, and capture later" setting: "The setting is not applied to In-Person Payments (please note that In-Person Payments should be captured within 2 days of authorization)."

must-be-captured

#### Testing instructions

* set `card_present_enabled` meta to 1 for the current store
* go to WooCommerce->Settings->Payments
* in "Transaction preferences" section, under "Issue an authorization on checkout, and capture later" field you should see a new help text: "Note: This setting is ignored by in-person payments. In-person payments authorization is valid for 48 hours."
* the text should not be present if in-person payments are not enabled.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

File: client/settings/transactions-and-deposits/test/index.test.js [hunk 1]
```diff
@@ -28,6 +30,7 @@ describe( 'TransactionsAndDeposits', () => {
 		useManualCapture.mockReturnValue( [ false, jest.fn() ] );
 		useGetSavingError.mockReturnValue( null );
 		useSavedCards.mockReturnValue( [ false, jest.fn() ] );
+		useCardPresentEligible.mockReturnValue( [ false ] );
```
Comment (anu-rock) [inline client/settings/transactions-and-deposits/test/index.test.js:33]: It'll help to also test a scenario where `useCardPresentEligible` returns true. The message displayed will vary in both cases. Your expectation will be similar to what we see here:

https://github.com/Automattic/woocommerce-payments/blob/develop/client/settings/transactions-and-deposits/test/index.test.js#L97-L101

File: client/data/settings/hooks.js [hunk 2]
```diff
@@ -20,6 +20,16 @@ export const useSavedCards = () => {
 	return [ isSavedCardsEnabled, updateIsSavedCardsEnabled ];
 };
 
+export const useCardPresentEligible = () => {
+	const { updateIsCardPresentEligible } = useDispatch( STORE_NAME );
+
+	const isCarPresentEligible = useSelect( ( select ) => {
```
Comment (anu-rock) [inline client/data/settings/hooks.js:26]: I think there's a typo here. Perhaps you meant `isCardPresentEligible`?

File: client/settings/transactions-and-deposits/index.js [hunk 3]
```diff
@@ -41,17 +42,7 @@ const TransactionsAndDeposits = () => {
...
-	// 	'When enabled, users will be able to pay with a saved card during checkout. ' +
-	// 	'Card details are stored in our platform, not on your store.' +
-	// 	'Note: This setting is ignored by in-person payments. In-person payments authorization is valid for 48 hours.',
-	// 	'woocommerce-payments'
-	// ) : __(
-	// 	'When enabled, users will be able to pay with a saved card during checkout. ' +
-	// 	'Card details are stored in our platform, not on your store.',
-	// 	'woocommerce-payments'
-	// );
+	const [ isCarPresentEligible ] = useCardPresentEligible();
```
Comment (anu-rock) [inline client/settings/transactions-and-deposits/index.js:45]: We may have another typo here ;)

File: includes/class-wc-payment-gateway-wcpay.php [hunk 4]
```diff
@@ -492,6 +492,25 @@ public function is_saved_cards_enabled() {
...
 
+	/**
+	 * Check if account is eligible for card present.
+	 *
+	 * @param null $empty_value - Default return value.
+	 * @return bool
+	 */
+	public function is_card_present_eligible( $empty_value = false ) {
+		try {
+			if ( ! $this->is_connected() ) {
```
Comment (anu-rock) [inline includes/class-wc-payment-gateway-wcpay.php:503]: This does not seem like an error-prone piece of code. We may want to take it out of the `try` block?

**Comment by anu-rock** [comment]
@dpaun1985 Great work with React code 🥳  I've left some initial feedback, mostly minor comments.

**Comment by allendav** [comment]
We mustn't tie this to cardPresentEligible. That flag was only meant for use during the beta. See pc2DNy-vL-p2

Let's just update the text for all, e.g.:

"Online charges must be captured on the order details screen within 7 days of authorization and In-Person Payments must be captured within 2 days, otherwise the authorization and order will be canceled."

cc @ctarda @xiaoshen-stripe

**Comment by anu-rock** [comment]
@allendav We've left the `card_present_eligible` check on purpose after discussing internally. It's true that the `card_present_enabled` CLI flag is no longer needed as we approach IPP's general availability. But since IPP will be rolled out in various geographies in phases, it's important that we do not present IPP-specific stuff to not yet eligible folks. We'll very soon change how the `card_present_eligible` flag is calculated internally in our PHP code. Perhaps we'll stop relying on the `card_present_enabled` account metadata flag and instead start depending on the merchant's country or other related information.

> Also added to support docs just now here: https://docs.woocommerce.com/document/getting-started-with-in-person-payments-with-woocommerce-payments/#authorizations-and-captures

Thanks 🙌 

cc @kalessil

File: includes/class-wc-payment-gateway-wcpay.php (hunk 4 above)
Comment (dpaun1985) [reply includes/class-wc-payment-gateway-wcpay.php]: good hint. moved it outside the try block

File: client/settings/transactions-and-deposits/index.js (hunk 3 above)
Comment (dpaun1985) [reply client/settings/transactions-and-deposits/index.js]: indeed 😄 . changed.

File: client/settings/transactions-and-deposits/test/index.test.js (hunk 1 above)
Comment (dpaun1985) [reply client/settings/transactions-and-deposits/test/index.test.js:33]: just added some tests.

**Comment by allendav** [comment]
@anu-rock wrote:

> We've left the card_present_eligible check on purpose after discussing internally. It's true that the card_present_enabled CLI flag is no longer needed as we approach IPP's general availability. But since IPP will be rolled out in various geographies in phases, it's important that we do not present IPP-specific stuff to not yet eligible folks. We'll very soon change how the card_present_eligible flag is calculated internally in our PHP code.

I see the proposed changes in server 1201 and I agree and I think its lovely. Thank you!

File: includes/admin/class-wc-rest-payments-settings-controller.php [hunk 5]
```diff
@@ -198,6 +198,7 @@ public function get_settings(): WP_REST_Response {
 				'payment_request_button_type'       => $this->wcpay_gateway->get_option( 'payment_request_button_type' ),
 				'payment_request_button_theme'      => $this->wcpay_gateway->get_option( 'payment_request_button_theme' ),
 				'is_saved_cards_enabled'            => $this->wcpay_gateway->is_saved_cards_enabled(),
+				'is_card_present_eligible'          => $this->wcpay_gateway->is_card_present_eligible(),
```
Comment (kalessil) [inline includes/admin/class-wc-rest-payments-settings-controller.php:201]: Looks like we have mix of tabs/spaces in the array formatting, let's stick to the same style as in the first items of the array.

File: includes/class-wc-payment-gateway-wcpay.php [hunk 6]
```diff
@@ -492,6 +492,26 @@ public function is_saved_cards_enabled() {
...
 
+	/**
+	 * Check if account is eligible for card present.
+	 *
+	 * @param false $empty_value - Default return value.
+	 * @return bool
+	 */
+	public function is_card_present_eligible( $empty_value = false ) {
+
+		if ( ! $this->is_connected() ) {
```
Comment (kalessil) [inline includes/class-wc-payment-gateway-wcpay.php:503]: Looking into `is_card_present_eligible` implementation, this if-construct looks like a redundant. Can we get rid of it (I also believe we'll also simplify mocking in UTs)?

File: client/settings/transactions-and-deposits/index.js [hunk 7]
```diff
@@ -68,11 +70,23 @@ const TransactionsAndDeposits = () => {
...
-						'woocommerce-payments'
-					) }
+					help={
+						<span>
+							{ __(
+								'Charge must be captured on the order details screen within 7 days of authorization, ' +
+									'otherwise the authorization and order will be canceled.',
+								'woocommerce-payments'
+							) }
+							<br />
```
Comment (kalessil) [inline client/settings/transactions-and-deposits/index.js:80]: Let's skip this `<br/>` :thinking:

File: client/settings/transactions-and-deposits/index.js [hunk 8]
```diff
@@ -68,11 +70,23 @@ const TransactionsAndDeposits = () => {
...
+							{ isCardPresentEligible
+								? __(
+										' Note: This setting is ignored by in-person payments. ' +
```
Comment (kalessil) [inline client/settings/transactions-and-deposits/index.js:83]: Suggesting wording: `The setting is not applied to In-Person Payments (please note that In-Person Payments should be captured within 2 days of authorization).`

**Review by kalessil — CHANGES_REQUESTED** [review]
Great work here @dpaun1985, the PR works as expected !
I left a few comments though, which we need to address before moving forward.

File: client/settings/transactions-and-deposits/index.js (hunk 8 above)
Comment (allendav) [reply client/settings/transactions-and-deposits/index.js:83]: I'm confused. If the setting is not applied to in-person payments (I thought it was), why would the merchant ever need to capture an in-person payment manually? Isn't this setting applicable to online AND in-person payments?

Comment (dpaun1985) [reply client/settings/transactions-and-deposits/index.js]: @allendav, at this point this setting isn't applicable for IPP. The charge is captured at the moment the client pays.

Comment (allendav) [reply client/settings/transactions-and-deposits/index.js]: Ah. Perfect then :)

**Comment by anu-rock** [comment]
LGTM :shipit:

**Review by kalessil — APPROVED** [review]
LGTM !

**Pull request merged**